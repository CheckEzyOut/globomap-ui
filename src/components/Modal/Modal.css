.modal {
  display: flex;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.75);
  z-index: 99;
}

.modal .close-modal-btn {
  position: absolute;
  top: 20px;
  right: 20px;
}

.modal .modal-loading {
  font-size: 64px;
  color: rgba(255, 255, 255, 0.3);
}

.modal .modal-content {
  z-index: 101;
}
