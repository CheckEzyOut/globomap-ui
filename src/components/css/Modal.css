.modal {
  display: flex;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.75);
  z-index: 99;
}

.modal .close-modal-btn {
  position: absolute;
  top: 20px;
  right: 20px;
  color: white;
  font-size: 18px;
}

.modal .close-modal-btn:hover {
  color: #333;
}

.modal .modal-loading {
  font-size: 64px;
  color: rgba(255, 255, 255, 0.3);
}

.modal .modal-content {  }
