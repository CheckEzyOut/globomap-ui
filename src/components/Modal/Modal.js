/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import './Modal.css';

class Modal extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      content: this.props.content
    }
  }

  render() {
    return this.props.visible
            ? <div className="modal" onClick={(e) => this.props.closeModal()}>
                <button className="close-modal-btn topcoat-button--quiet"
                  onClick={(e) => this.props.closeModal()}>
                  <i className="fa fa-times"></i>
                </button>
                <div className="modal-content" onClick={(e) => e.stopPropagation()}>
                  {this.props.content !== null
                    ? this.props.content
                    : <i className="modal-loading fa fa-cog fa-spin fa-2x fa-fw"></i>}
                </div>
              </div>
            : null
  }
}

export default Modal;
