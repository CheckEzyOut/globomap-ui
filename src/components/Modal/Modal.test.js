import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from './Modal';

describe("Modal", function() {

  it('should render Modal component', () => {
    const component = shallow(<Modal />);
    expect(component).toMatchSnapshot();
  });

});
