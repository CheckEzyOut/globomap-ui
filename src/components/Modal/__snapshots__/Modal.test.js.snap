// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render Modal component 1`] = `""`;
